DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs

```diff
@@ -2182,4 +1569,198 @@ internal class ContextMenuCommand
+                var subCommand = subCommandParent.Options.FirstOrDefault(op => op.Name.Equals(commandOneName,StringComparison.OrdinalIgnoreCase));
+                var discordEmbed = new DiscordEmbedBuilder
+                {
+                    Title = "Help",
+                    Description = $"{Formatter.InlineCode(subCommand.Name)}: {subCommand.Description ?? "No description provided."}"
+                };
+                if (subCommand.Options is not null)
+                {
+                    var commandOptions = subCommand.Options.ToList();
+                    var sb = new StringBuilder();
+
+                    foreach (var option in commandOptions)
+                        sb.Append('`').Append(option.Name).Append(" (").Append(")`: ").Append(option.Description ?? "No description provided.").Append('\n');
+
+                    sb.Append('\n');
+                    discordEmbed.AddField("Arguments", sb.ToString().Trim(), false);
+                }
+                await ctx.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource,
+                    new DiscordInteractionResponseBuilder().AddEmbed(discordEmbed).AsEphemeral(true));
+            }
+            else
+            {
+                DiscordApplicationCommand command = applicationCommands.FirstOrDefault(cm => cm.Name.Equals(commandName, StringComparison.OrdinalIgnoreCase));
+                if (command is null)
+                {
+                    await ctx.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, new DiscordInteractionResponseBuilder()
+                            .WithContent($"No command called {commandName} in guild {ctx.Guild.Name}").AsEphemeral(true));
+                    return;
+                }
+                else
+                {
+                    var discordEmbed = new DiscordEmbedBuilder
+                    {
+                        Title = "Help",
+                        Description = $"{Formatter.InlineCode(command.Name)}: {command.Description ?? "No description provided."}"
+                    };
+                    if (command.Options is not null)
+                    {
+                        var commandOptions = command.Options.ToList();
+                        var sb = new StringBuilder();
+
+                        foreach (var option in commandOptions)
+                            sb.Append('`').Append(option.Name).Append(" (").Append(")`: ").Append(option.Description ?? "No description provided.").Append('\n');
+
+                        sb.Append('\n');
+                        discordEmbed.AddField("Arguments", sb.ToString().Trim(), false);
+                    }
+                    await ctx.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource,
+                        new DiscordInteractionResponseBuilder().AddEmbed(discordEmbed).AsEphemeral(true));
+                }
+            }
```

Comment:
axe the else -- indent left

Reply:
kjhz8iohzgiophzopö

Reply:
that's @villChurch s job

Reply:
@Lulalaby Does @villChurch have the ability to touch this branch seeing that it's on the ACTUAL repo

Reply:
mhh one sec